RPG/Src/XCOM2RPGOverhaul/Classes/X2TemplateHelper_RPGOverhaul.uc

```diff
@@ -482,20 +490,28 @@ static function PatchWeapons()
 						WeaponTemplate.NumUpgradeSlots = default.DefaultWeaponUpgradeSlots;
 						break;
 					case 'shotgun':
+						if (default.bAddWeaponsChanges)
+						{					
 						AddAbilityToWeaponTemplate(WeaponTemplate, 'ShotgunDamageModifierCoverType');
 						AddAbilityToWeaponTemplate(WeaponTemplate, 'ShotgunDamageModifierRange');
 						
 						WeaponTemplate.CritChance += default.ShotgunCritBonus;
 						WeaponTemplate.Aim += default.ShotgunAimBonus;
+						}
 						WeaponTemplate.NumUpgradeSlots = default.DefaultWeaponUpgradeSlots;
 						break;
 					case 'cannon':
+						if (default.bAddFullAutoFire)
+						{					
 						AddAbilityToWeaponTemplate(WeaponTemplate, 'FullAutoFire', true);
+						}
```

Comment:
same here i suggest creating a new variable like
`bPatchCannons`
and include
```
AddAbilityToWeaponTemplate(WeaponTemplate, 'Suppression', true);
AddAbilityToWeaponTemplate(WeaponTemplate, 'HeavyWeaponMobilityPenalty', true);
```
in the if